src/main/scala/com.snowplowanalytics.iglu/ctl/commands/TableCheck.scala

```diff
@@ -34,216 +32,291 @@ import com.snowplowanalytics.iglu.ctl.{Command, Common, Failing, File, Server, S
-    case CommentProblem(schemaKey, message) =>
-      s"Table for ${schemaKey.toSchemaUri}. $message"
+    Storage.commonColumns ++ customColumns
   }
 
-  implicit val aggregatedTableCheckResultsShow: Show[AggregatedResult] = Show.show {
-    case AggregatedResult(matched, unmatched, notDeployed, commentProblems) =>
-      s"""
-         |${createSection("Not deployed:", notDeployed.map((_: Result).show).mkString("\n"))}
-         |
-         |${createSection("Matched:", matched.map((_: Result).show).mkString("\n"))}
-         |
-         |${createSection("Table comments:", commentProblems.map((_: Result).show).mkString("\n"))}
-         |
-         |${createSection("Unmatched:", unmatched.map((_: Result).show).mkString("\n"))}
-         |
-         |Unmatched: ${unmatched.length}, Comment issues: ${commentProblems.length}, Matched: ${matched.length}, Not Deployed: ${notDeployed.length}
-       """.stripMargin.split("\n").filter(_.nonEmpty).mkString("\n")
+  private[ctl] def verifyExistingStorage(latestSchemaKey: SchemaKey,
+                                         existingComment: Option[Storage.Comment],
+                                         existingColumns: List[Column],
+                                         expectedColumns: List[Column]): Result = {
+    existingColumns match {
+      case Nil => TableNotDeployed(latestSchemaKey)
+      case _ =>
+        val columnIssues = detectColumnIssues(existingColumns, expectedColumns)
+        val commentIssues = checkComment(existingComment, latestSchemaKey).toList
+        
+        NonEmptyList.fromList(commentIssues ::: columnIssues) match {
+          case Some(discoveredIssues) =>
+            TableUnmatched(latestSchemaKey, discoveredIssues, expectedColumns, existingColumns)
+          case None =>
+            TableMatched(latestSchemaKey) 
+        }
+    }
   }
 
-  /**
-    * Primary method of table-check command. Checks whether table of the given schema
-    * has expected structure which is determined with last and previous versions of
-    * the given schema
-    */
-  def process(command: Command.TableCheck, httpClient: Client[IO])(implicit cs: ContextShift[IO], t: Timer[IO]): FinalResult = {
-    val stream = for {
-      resolvedDbConfig <- Stream.eval(Storage.resolveDbConfig(command.storageConfig))
-      storage <- Stream.resource(Storage.initialize[IO](resolvedDbConfig)).translate[IO, Failing](Common.liftIO)
-      res <- command.tableCheckType match {
-        case Command.SingleTableCheck(resolver, schema) =>
-          Stream.eval(tableCheckSingle(resolver, schema, storage, command.dbSchema))
-        case Command.MultipleTableCheck(igluServerUrl, apiKey) =>
-          tableCheckMultiple(igluServerUrl, apiKey, storage, command.dbSchema, httpClient)
+  private def detectColumnIssues(existingColumns: List[Column], expectedColumns: List[Column]): List[TableIssue] = {
+    expectedColumns.map(_.some).zipAll(existingColumns.map(_.some), None, None)
```

Comment:
Order of columns should not be important. New ShredModel would shuffle them around (compared to the old version), but since loader addresses them explicitly it is not an issue.

Reply:
it should not